pkg/assembler/backends/inmem/certifyVuln.go

```diff
@@ -212,6 +213,21 @@ func (c *demoClient) CertifyVuln(ctx context.Context, filter *model.CertifyVulnS
 			foundOne = true
 		}
 	}
+	if !foundOne && filter != nil && filter.Vulnerability != nil &&
+		filter.Vulnerability.NoVuln != nil && *filter.Vulnerability.NoVuln {
+
+		exactVuln, err := c.exactVulnerability(&model.VulnerabilitySpec{
+			Type:            ptrfrom.String(noVulnType),
+			VulnerabilityID: ptrfrom.String(""),
+		})
+		if err != nil {
+			return nil, gqlerror.Errorf("%v :: %v", funcName, err)
+		}
+		if exactVuln != nil {
+			search = append(search, exactVuln.certifyVulnLinks...)
+			foundOne = true
+		}
+	}
 	if !foundOne && filter != nil && filter.Vulnerability != nil {
 		exactVuln, err := c.exactVulnerability(filter.Vulnerability)
```

Comment:
do we need to check here that we skip novulns

Reply:
addded check in the resolver such that if `novuln` is set to false, the type cannot be set to "novuln"

Reply:
added a check here and in vulnerabilities to check also.

Reply:
```
query CertifyNoVuln {
  CertifyVuln(certifyVulnSpec: {
    vulnerability: {noVuln: false, type: "novuln"}
  }) {
    ...allCertifyVulnTree
  }
}
```
will return in error:
```
{
  "errors": [
    {
      "message": "novuln boolean set to false, cannot specify vulnerability type to be novuln",
      "path": [
        "CertifyVuln"
      ]
    }
  ],
  "data": null
}
```